packages/svelte/src/remix-router-svelte.ts

```diff
@@ -0,0 +1,181 @@
+import {
+  Action as NavigationType,
+  resolveTo,
+  type Fetcher,
+  type FormEncType,
+  type FormMethod,
+  type Navigation,
+  type Router,
+  type Location,
+} from "@remix-run/router";
+import { onDestroy, SvelteComponent } from "svelte";
+import { derived, writable, type Readable, type Writable } from "svelte/store";
+import { getRouteContext, getRouterContext } from "./contexts";
+import { getFormSubmissionInfo, type SubmitOptions } from "./dom";
+
+type FetcherWithComponents<TData> = Fetcher<TData> & {
+  Form: SvelteComponent;
+  key: string;
+  // TODO: abstract via useSubmitImpl
+  submit(target: SubmitTarget, options?: SubmitOptions): void;
+  load: (href: string) => void;
+};
+
+type SubmitTarget =
+  | HTMLFormElement
+  | HTMLButtonElement
+  | HTMLInputElement
+  | FormData
+  | URLSearchParams
+  | { [name: string]: string }
+  | null;
+
+export function useLoaderData() {
+  let ctx = getRouteContext();
+  return useRouteLoaderData(ctx.id);
+}
+
+export function useRouteLoaderData(routeId: string) {
+  let ctx = getRouterContext();
+  // return derived(ctx.state, ({ loaderData }) => loaderData[routeId] as unknown);
+  // it may cause bugs to return non-reactive state here, not sure. this works for now.
+  return getStoreSnapshot(ctx.state).loaderData[routeId] as unknown;
+}
+
+export function useLocation(): Readable<Location> {
+  let ctx = getRouterContext();
+  return derived(ctx.state, ({ location }) => location);
```

Comment:
Any idea if these suffer the same issue that `useLoaderData` did using the `derived` store instead of `getStoreSnapshot`?